pallene/types.lua

```diff
@@ -26,14 +26,16 @@ declare_type("T", {
         "field_names", -- same order as the source type declaration
         "field_types", -- map { string => types.T }
     },
+    Module = {},
```

Comment:
Can we find a way to not have a module type? I'm happy that we recently got rid of the "fake" types `Void` and `Module` and I'm not looking forward to putting them back...

Perhaps by treating the `require` as special syntax instead of as a regular function call. Or maybe by saying that it returns `any`.

Reply:
@hugomg using `any` instead of modules works fine, as well.
Just committed this change now.

Reply:
Does that mean the user won't be able to shadow `require` anymore?
perhaps it could be a contextual keyword like `async` in Javascript.

Reply:
As of now, the user is still able to shadow `require`.
I think that's actually a good test case idea.